simple_app/utils.py

```diff
@@ -790,79 +826,81 @@ def write_file(results: pd.DataFrame) -> str:
-    nowtime = strftime("%Y-%m-%d--%H-%M-%S", localtime())
-    fname = 'simple_app/tmp/userquery-' + nowtime + '.zip'
+    csv_dict: Dict[str, StringIO] = {}
     for key, df in resultsdict.items():
-        csvname = fname.replace('.zip', f'_{key}.csv')
-        df.to_csv(csvname, index=False)
-    with ZipFile(fname, 'w') as zipper:
-        for dirname, subdirname, filenames in os.walk('simple_app/tmp'):
-            for csvname in filenames:
-                if nowtime not in csvname or 'csv' not in csvname:
-                    continue
-                fpath = os.path.join(dirname, csvname)
-                zipper.write(fpath, os.path.basename(csvname))
-    return fname
+        csv_data = StringIO()
+        df.to_csv(csv_data, index=False)
+        csv_data.seek(0)
+        csv_dict[key] = csv_data
+    zip_mem = BytesIO()
+    with ZipFile(zip_mem, 'w') as zipper:
+        for key, csv_data in csv_dict.items():
+            zipper.writestr(f"{key}.csv", csv_data.getvalue())
+    zip_mem.seek(0)
+    return zip_mem
 
 
-def write_fitsfiles(fitsfiles: List[str]) -> str:
+# noinspection PyTypeChecker
+# this is because pycharm isn't the smartest
+def write_fitsfiles(fitsfiles: List[str]) -> BytesIO:
     """
-    Creates a csv file ready for download
+    Creates a zip file containing multiple csvs ready for download
 
     Parameters
     ----------
-    fitsfiles: List[str]
-        The urls to the fits files
+    fitsfiles
+        The list of fits files for this object
 
     Returns
     -------
-    fname: str
-        The filename
+    zip_mem
+        The zipped file in memory
     """
-    [os.remove('simple_app/tmp/' + f) for f in os.listdir('simple_app/tmp/') if 'README' not in f]  # clear out
-    nowtime = strftime("%Y-%m-%d--%H-%M-%S", localtime())
-    fname = 'simple_app/tmp/userquery-' + nowtime + '.zip'
+    fits_dict: Dict[str, BytesIO] = {}
     for i, fitsfile in enumerate(fitsfiles):
-        fitsname = fname.replace('.zip', f'_{i}.{fitsfile.split(".")[-1]}')
-        with open(fitsname, 'wb') as handle:
-            r = requests.get(fitsfile)
-            for data in r.iter_content():
-                handle.write(data)
-    with ZipFile(fname, 'w') as zipper:
-        for dirname, subdirname, filenames in os.walk('simple_app/tmp'):
-            for fitsname in filenames:
-                if nowtime not in fitsname or 'zip' in fitsname:
-                    continue
-                fpath = os.path.join(dirname, fitsname)
-                zipper.write(fpath, os.path.basename(fitsname))
-    return fname
+        fits_mem = BytesIO()
+        try:
+            with fits.open(fitsfile) as hdulist:
+                hdulist.writeto(fits_mem)
+        except (OSError, fits.verify.VerifyError):  # spectra which can't be loaded properly
+            continue
+        fits_mem.seek(0)
+        fits_dict[f"spectra_{i}_" + os.path.basename(fitsfile)] = fits_mem
```

Comment:
Also, as a thought, the code I'm using could be relevant for testing spectra ingests, e.g. https://github.com/SIMPLE-AstroDB/SIMPLE-db/issues/287, as it used requests to physically download the fits files and re-output them. If that runs into issues, the spectra needs fixing.